lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -48,6 +48,35 @@ function wp_theme_has_theme_json() {
 	}
 }
 
+if ( ! function_exists( 'wp_theme_get_css_selector_for_block' ) ) {
+	/**
+	 * Lookup a CSS selector for the block provided, and return it if it exists.
+	 *
+	 * @param string $block_name The name of the block to lookup the CSS selector for.
+	 *
+	 * @return string|null the CSS selector for the block.
+	 */
+	function wp_theme_get_css_selector_for_block( $block_name ) {
```

Comment:
The theme and the block are unrelated, so it seems better to rename this to something along the lines of `wp_block_get_css_selector`.

Note [this conversation](https://github.com/WordPress/gutenberg/issues/45194). A block has more than one selector. This method should be able to retrieve any of the selectors provided by the block.

Reply:
It'd be good to update `get_blocks_metadata` to use this method as well. It'd give us feedback whether this is good in practice or not.

For example, I haven't really dug into this, but just thinking about reusing this method at `get_blocks_metadata` it struck me that an alternative to this PR could be adding a `get_css_selector` method to the `WP_Block_Type` class instead.

Reply:
> The theme and the block are unrelated, so it seems better to rename this to something along the lines of wp_block_get_css_selector.

Naive question, should we be prefixing this with `gutenberg_` prior to it landing in core?

> Note this conversation. A block has more than one selector. This method should be able to retrieve any of the selectors provided by the block.

+1 on this comment. Not only can blocks have selectors on a root and per-block-support basis, in the near future we plan on allowing custom selectors for individual styles for example on the color's background.

The point above probably illustrates we'll get even more benefit in future from reusing the results of this PR.

> an alternative to this PR could be adding a get_css_selector method to the WP_Block_Type class instead.

Sounds good to me 👍